assets/model_monitoring/components/src/action_analyzer/contracts/action_detectors/metrics_violation_index_action_detector.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Metrics violation index action detector class."""
+
+from action_analyzer.contracts.action_detector import ActionDetector
+from action_analyzer.contracts.action import Action
+from typing import Dict
+import pandas
+
+SUPPORTED_METRICS = ["Fluency", "Coherence", "Relevance", "Groundedness", "RetrievalRelevance"]
+
+
+class MetricsViolationIndexActionDetector(ActionDetector):
+    """Metrics violation index action detector class."""
+
+    def __init__(self,
+                 workspace_connection_arm_id: str,
+                 model_deployment_name: str,
+                 aml_deployment_id: str,
+                 metrics_violation_thresholds: Dict[str, float],
+                 metrics_violation_passing_rate: Dict[str, float],
+                 correlation_test_method: str,
+                 correlation_test_pvalue_threshold: float,
+                 action_max_positive_sample_size: int,
+                 llm_summary_enabled: str) -> None:
+        """Create a metrics violation index action detector.
+
+        Args:
+            workspace_connection_arm_id(str): azureml workspace connection arm id for llm.
+            model_deployment_name(str): model deployment name of the connection.
+            aml_deployment_id(str): the azureml deployment id of the llm application.
+            metrics_violation_thresholds(Dict[str, float]): metrics violation thresholds dict, key is metric name, value is threshold for that metric.
```

Comment:
regarding these two paramters metrics_violatin_thresholds and metrics_violation_passing_rate. my thoughts:
- there should be a upstream component that does
-- e2e metric calculation
-- identify violated metrics based on user set threshold when they created MoMo

that upstream component can either by GSQ, or some code in action analyzer (for the case when action analyzer is not linked after GSQ).

so to the detector, what it need is
- list of metrics that has violations.
- the threshold of the violation per each metric. (this is also debatable depends on how do we use it in the detector, considering now MoMo user only sets threshold on the average score).

it should not need the passing_rate